src/idl_gen_js_ts.cpp

```diff
@@ -1222,6 +1223,17 @@ class JsTsGenerator : public BaseGenerator {
     obj_api_unpack_func = unpack_func + "\n\n" + unpack_to_func;
   }
 
+  bool CanCreateFactoryMethod(const StructDef &struct_def) {
+    // to preserve backwards compatibility, we allow the first field to be a struct
```

Comment:
@aardappel @krojew Is this valid? If we use the `createX()` method when a struct is the first field, the buffer fails verification in other languages, but succeeds if we create the struct after the generated call to `startX()`.

Reply:
Hard to say what the reason was, since it was so long ago. But we're making backwards-incompatible changes anyway so this can be removed if it causes problems.

Reply:
Thanks! I believe it should be removed since as far as I understand the buffer will always be invalid even if the struct is just the first field.